### PR TITLE
Docs, form model

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/guide/en/form-model.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+Form model is an abstraction over HTML forms. While you can use [forms](https://github.com/yiisoft/form) directly, 
+it can be more convenient to define form in object-oriented style. With this approach, form fields are defined as class
+properties. Besides describing form data, form model also handles presentation and validation aspects.
+
+To define a form model, create a class extending from `Yiisoft\FormModel\FormModel`.
+
+## Properties
+
+Form model properties are defined as class properties. You can also add getters and setters for each of them.
```
Comment (vjik) [inline docs/guide/en/form-model.md:11]: Why need getters/setters? We can just make properties public.

File: docs/guide/en/form-model.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        #[Length(max: 255)]
+        private string $name,
+        #[Collection(Post::class)]
+        #[Each([new Nested(Post::class)])]
+        private array $posts = [],
+    ) {
+    }
+}
+
+final class Post
```
Comment (vjik) [inline docs/guide/en/form-model.md:328]: This is not form.

**Review by vjik — CHANGES_REQUESTED** [review]

File: docs/guide/en/form-model.md (hunk 1 above)
Comment (samdark) [reply docs/guide/en/form-model.md:11]: `can` already means that it's not mandatory.

File: docs/guide/en/form-model.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+> Static properties are not included in form model's set of properties. On attempt to get such property's value,
+> `Yiisoft\FormModel\Exception\StaticObjectPropertyException` exception will be thrown.
+
+## Meta data
+
+Form model's meta data includes:
+
+- Property labels - the labels associated with each input (visible to the end user). When not set, they will be 
+automatically generated using inflector from the [strings](https://github.com/yiisoft/strings) package.
```
Comment (samdark) [inline docs/guide/en/form-model.md:80]: Worth linking to exact method.

File: docs/guide/en/form-model.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    #[Length(min: 8, lessThanMinMessage: 'Is too short.')]
+    private ?string $password = null;    
+    
+    // ...    
+}
+```
+
+### `Safe` rule
+
+This package also provides `Safe` rule that marks a model property as safe for filling.
```
Comment (samdark) [inline docs/guide/en/form-model.md:275]:
```suggestion
This package also provides `Safe` rule that marks a model property as safe for filling from user input.
```

File: docs/guide/en/form-model.md (hunk 3 above)
Comment (arogachev) [reply docs/guide/en/form-model.md:80]: To where exactly? Guide doesn't have dedicated section for method, only for group of methods or `Inflector` in general. Or do you mean to add link to the method's source code?

Comment (samdark) [reply docs/guide/en/form-model.md:53]: I've meant source code.

File: docs/guide/en/nested-form-models.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Nested form models
```
Comment (samdark) [inline docs/guide/en/nested-form-models.md:1]: It's confusing that we call the guide “Nested form models” and then immediately tell that actually these are not form models.

How about naming it differently, such as “Handling nested data and nested forms”?

Comment (arogachev) [reply docs/guide/en/nested-form-models.md:1]: I suggest just "Handling nested data" as more common name then.

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Pull request merged**